web/backend/src/Server.ts

```diff
@@ -154,49 +185,58 @@ app.post('/api/logout', (req, res) => {
 // As the user is logged on the app via this express but must also be logged in
 // the react. This endpoint serves to send to the client (actually to react)
 // the information of the current user.
-app.get('/api/personal_info', (req, res) => {
-  res.set('Access-Control-Allow-Origin', '*');
-  if (req.session.userid) {
-    res.json({
-      sciper: req.session.userid,
-      lastname: req.session.lastname,
-      firstname: req.session.firstname,
-      role: req.session.role,
-      islogged: true,
-    });
-  } else {
-    res.json({
-      sciper: 0,
-      lastname: '',
-      firstname: '',
-      role: '',
-      islogged: false,
-    });
-  }
-});
-
-function isAuthorized(roles: string[], req: express.Request): boolean {
-  if (!req.session || !req.session.userid) {
-    return false;
+function setMapAuthorization(list: string[][]) {
+  const m = new Map<String, Array<String>>();
+  for (let i = 0; i < list.length; i += 1) {
+    for (let j = 1; j < list[0].length; j += +2) {
+      console.log(list[i][j]);
+      if (m.has(list[i][j])) {
+        m.get(list[i][j])?.push(list[i][j + 1]);
+      } else {
+        m.set(list[i][j], [list[i][j + 1]]);
+      }
+    }
   }
-
-  const { role } = req.session;
-
-  return roles.includes(role as string);
+  console.log(m);
+  return m;
 }
+app.get('/api/personal_info', (req, res) => {
+  let m = new Map<String, Array<String>>();
+  enf.getFilteredPolicy(0, String(req.session.userid)).then((list) => {
+    m = setMapAuthorization(list);
```

Comment:
If the user is not logged you can avoid calling `setMapAuthorization` and return an empty map.

Reply:
Fixed !